cloudevents/abstract/event.py

```diff
@@ -0,0 +1,94 @@
+import typing
+from typing import TypeVar
+
+
+class CloudEvent:
+    """
+    The CloudEvent Python wrapper contract exposing generically-available
+    properties and APIs.
+
+    Implementations might handle fields and have other APIs exposed but are
+    obliged to follow this contract.
+    """
+
+    @classmethod
+    def create(
+        cls,
+        attributes: typing.Dict[str, typing.Any],
+        data: typing.Optional[typing.Any],
+    ) -> "AnyCloudEvent":
+        """
+        Factory function.
+        We SHOULD NOT use the __init__ function as the factory because
+        we MUST NOT assume how the __init__ function looks across different python
+        frameworks
+        """
+        raise NotImplementedError()
+
+    @classmethod
+    def get_attributes(cls, event: "CloudEvent") -> typing.Dict[str, typing.Any]:
+        """
+        :return: Attributes of this event.
+
+        You MUST NOT mutate this dict.
+        Implementation MAY assume the dict will not be mutated.
+        """
+        raise NotImplementedError()
+
+    @classmethod
+    def get_data(cls, event: "CloudEvent") -> typing.Optional[typing.Any]:
+        """
+        :return: Data value of the  event.
+        You MUST NOT mutate this dict.
+        Implementation MAY assume the dict will not be mutated.
+        """
+        raise NotImplementedError()
```

Comment:
It is my way of handling the pydantic edge case for the data field. I tried making an alias but it messes everything up.
I don't see any reason for forcing any data attributes or member functions for our concrete event type.
All I want is the generic access of the values im interested in especially if this is for a internal use only. 

I think I may make it private to prevent confusion.

Reply:
I still don't get the reasoning. Just `my way of handling is not a good enough reason.

I don't see a reason why `get_data(event)` can be anyhow different from `event.get_data`. The latter one looks and feels more natural. Moreover, it is the event instance behavior to return some part of its state to the caller. Even if we make these APIs private (with `__`) or protected (with `_`), they still look very much like some abstract instance APIs, not class-level ones.

Reply:
And also, why don't we return a copy of the internal state instead of stating that one `MUST NOT` mutate the state? (that's in case we want this to be part of the public API).

Reply:
I have no problem to create protected event._get_data and event._get_attributes.
this will solve the confusion i was afraid of

 I thought about copying the data, but was afraid of the efficiency implication though it may be premature optimization.